### DISCUSSION TITLE
Cluster support

### DISCUSSION THREAD

**Discussion by sahara101** [ZoeyVid/NPMplus#849] [Q&A] [answered]
Hello! 

Is it possible to use NPMplus in a cluster/HA mode? 

Thanks!

**Comment by Zoey2936** [comment]
currently not, what would you like to exactly see?

**Comment by sahara101** [comment]
I would like to have it installed on two separate servers, if one goes down for whether reason, the second one should pick up. So keepalived, but the issue would be how to sync the config from master to slave

**Reply by Zoey2936** [reply, marked as answer]
you can setup two npmplus instances, which share the /data mount on a network share

**Reply by sahara101** [reply]
Yes that would work…but it is still a single point of failure :(

**Reply by Zoey2936** [reply]
yes I know, but if I built something like this into npmplus it would be the same, execpt that the files were not copied on a network share but instead by npmplus on a other server which would be insecure from a security perspective, or do you have a good idea?

**Reply by sahara101** [reply]
What I have tried...and failed was to configure a sql master-slave instance. For the non DB data I would have used incron/rsync. 
My issue that I did not solve was a network one. Reverse Proxy was using a macvlan in docker (because I also use traefik and the ports are then in use) and keepalived was using network_mode host. I could not make keepalived see the proxy port :(

**Reply by kvassss** [reply]
> you can setup two npmplus instances, which share the /data mount on a network share

Hi! I'm facing the same issue
I'm thinking about synchronizing configs and database using rsync(or even unison for two way synchronization), but I have a question about tls certificates
Is it possible a situation where both instances will try to update the certificate? Would this cause problems? If yes, is it possible to disable automatic certificate renewal for 1 instance?

**Reply by Zoey2936** [reply]
you could set this value very high for one instace: https://github.com/ZoeyVid/NPMplus/blob/da871559b265e5a17aedc0a8c875eee40e7996a2/compose.yaml#L60C8-L60C32

**Reply by kvassss** [reply]
> you could set this value very high for one instace: https://github.com/ZoeyVid/NPMplus/blob/da871559b265e5a17aedc0a8c875eee40e7996a2/compose.yaml#L60C8-L60C32

I'll give it a try
Thanks for the quick response
